src/anm/analytics.js

```diff
@@ -0,0 +1,79 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+            } else {
+                sendViaGif(data);
+            }
+        }
+        if (!supportSendBeacon) {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (data) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = animatronUrl + '?data=' + encodeURIComponent(data);
+    };
+    if (self.enabled) {
+        if (!supportSendBeacon) {
+            engine.onDocReady(event);
```

Comment:
The problem with `setInterval`, as I said, is that it could affect page performance, if `sendBeacon` is not supported. Especially canvas, especially if it will run once every second. Every five minutes + on `unload` — ok, probably.

Look at every `sendBeacon` example, they just [sum up and] push data _once_ in a large array and send it on `window.unload`. Unload will always perform. Just, if `sendBeacon` is not supported, sending data could "delay the unload or affect the the performance of the next navigation".